File: D2/1859_백만_장자_프로젝트/1859.py
T = int(input())

for test_case in range(1, T + 1):
    n = int(input())
    arr = list(map(int, input().split()))
    dp = [0] * n
    max_num = 0
    
    for i in range(n - 1, -1, -1):
        max_num = max(max_num, arr[i])
        dp[i] = max_num
        
    print(f'#{test_case} {sum(dp) - sum(arr)}')

File: D2/1859_백만_장자_프로젝트/1859.cpp
#include<iostream>
#include <vector>

using namespace std;

int main(int argc, char** argv)
{
	int test_case;
	int T;

	cin>>T;

	for(test_case = 1; test_case <= T; ++test_case) {
        int n;
        cin >> n;
        vector<int> v(n, 0);
        for(int i = 0, a; i < n; i++) cin >> v[i];
        
        int max_num = -1;
        long long v_total = 0, dp_total = 0;
        for(int i = n - 1; i >= 0; i--) {
            max_num = max(max_num, v[i]);
            v_total += v[i];
            dp_total += max_num;
        }
        
        cout << "#" << test_case << " " << dp_total - v_total << '\n';
    }
	return 0;
}